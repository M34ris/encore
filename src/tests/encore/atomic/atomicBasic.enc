active class Foo : Actor
  var i : int

  def init(i : int) : unit
     this.i = i
     println(this.i)
  end
  
  def increment() : unit
     this.i += 1
     println(this.i)
  end
end

active class Main
  def dummy() : unit
     println(42)
  end
          
  def main() : unit
     var foo = new Foo(11)
     atomic foo as bar in
       foo ! increment()
       bar ! increment()
       this.dummy()
     end
  end
end
