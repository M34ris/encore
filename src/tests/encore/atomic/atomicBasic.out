11
12
13
42